2015/04/Upgrade_Insecure.md

```diff
@@ -0,0 +1,102 @@
+a page does not support HTTPS or stops supporting HTTPS, the page author cannot
+ensure that the content is loaded securely or at all. Inevitably, moving to
+a secure origin causes problems with mixed content blocking if the page has
+third party content that doesn't yet support HTTPS, a problem which the spec
+does not address.
+
+### ISSUE: Same-Origin vs Cross-Origin Behavior Unclear in Examples
+
+Talking to other TAG members about the spec, it became apparent that some of us
+thought the spec only applied upgrades to same-origin requests. I attribute
+most of my confusion to the examples in Section 1.2. Example #1 uses the
+example of `<img src="http://example.com/image.png">` being upgraded on
+`https://example.com` and Example #2 explicitly says that `<a
+href="http://not-example.com/">Home</a>` will *not* be upgraded on
+`https://example.com`. It would be better if Example #1 explicitly said that
+a third-party origin like `not-example.com` is upgradeable in that context, so
+that readers don't generalize Example #2 to all requests.
+
+### CLARIFICATION: Wording in Terminology
+
+The wording "depend on the upgrade-insecure-requests mechanism" in Section 2 is
+unclear. It seems to mean something like, "the same with and without
+upgrade-insecure-requests" from context, but I'm not sure.
+
+### COMMENT: +1 for Issue #184
+
+https://github.com/w3c/webappsec/issues/184 seems like a good thing for
+improving the smoothness of the HTTP to HTTPS transition.
+
+### ISSUE: Need Example for Upgrade Insecure Navigations Set
+
+It would be nice to have an example of a CSP directive with an upgrade insecure
+navigations set in the draft.
+
+### ISSUE: Upgrade All Navigations
+
+Is there a way to specify that all navigations should be upgraded? That seems
+useful if a webmaster doesn't want to change their header every time they add
+a link and are confident all their links will support HTTPS.
+
+### ISSUE: Inconsistent Wording in 4.1?
+
+Section 4.1:
+> We will not upgrade cross-origin navigational requests, with the exception of
+> form submissions
+
+I'm confused now because the document otherwise states that cross-origin
+navigational requests will be upgraded if they are in the "upgrade insecure
+navigations set" for a context.
+
+### CLARIFICATION: Violation Reports for Inherited Policy
+
+As mentioned in 6.2, there is a security issue if a document is able to get
+violation reports for cross-origin nested documents (iframes, etc.) which
+inherit upgrade policy. So if a nested document does not specify its reporting
+endpoint, do all reports from the nested document get blocked?
+
+### IDEA: Cache/Pin Successful Upgrades
+
+Thinking about the broader goal of encrypting the web, it would be nice if
+user agents could remember which subresources have been successfully upgraded
+through this mechanism. That way, on a page that has not set the CSP header,
+the known-upgradeable subresources could be upgraded anyway.
```

Comment:
CSP pinning is scoped to a host (modulo include-subdomains), right? I was actually thinking that a resource could always be upgraded on any website once the UA knows that it is upgradeable.

Use case:
https://example.com sends the upgrade header and embeds `<script src="http://jquery.com/jquery.js">`. The UA notes that `jquery.com` has been upgraded successfully. In the future, the user visits https://not-example.com which also embeds `<script src="http://jquery.com/jquery.js">`. not-example.com does not send the CSP header, but the UA tries to upgrade jquery anyway. 

This mechanism seems kind of clumsy though; I'm more a fan of letting hosts signal that they should be upgraded regardless of where they are subresourced. (as described below)